be/src/storage/memtable.cpp

```diff
@@ -206,6 +206,10 @@ Status MemTable::finalize() {
                     _result_chunk = upserts;
                 }
             }
+            if (_keys_type == KeysType::PRIMARY_KEYS && !_vectorized_schema->sort_key_idxes().empty()) {
+                _chunk = _result_chunk;
+                _sort(true, true);
+            }
```

Comment:
if sort key is part of primary key,  there is no need to sort again?

Reply:
yes, optimized